_pytest/main.py

```diff
@@ -388,7 +392,10 @@ def _perform_collect(self, args, genitems):
         else:
             if rep.passed:
                 for node in rep.result:
-                    self.items.extend(self.genitems(node))
+                    if self.line and self.nearest_func_name:
```

Comment:
`line` and `nearest_func_name` are not very descriptive names in my opinion, I suggest we drop `self.line` altogether (it is not needed at this point) and keep only the name of the function, I suggest `self._filter_by_function_name` instead. Also I suggest to avoid creating the intermediary list and duplicating the `genitems` call (more to make the code neater than anything else):

```python
items = self.genitems(node)
if self._filter_by_function_name:
    items = (x for x in items if self.select_by_function_name in item.name)
self.items.extend(items)
```

Reply:
Fixed